web/src/components/register/register.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import './register.css'
+
+import TextField from '../textField/textField'
+import Button from '../button/button'
+import GenderRadio from '../genderRadio/genderRadio'
+import MajorRadio from '../majorRadio/majorRadio'
+
+class Register extends Component {
+  
+  constructor(props) {
+    super(props)
+    
+    this.state = {
+      name: "",
+      surname: "",
+      major: "",
+      gender: "",
+      age: "",
+      valid: {
+        name: null,
+        surname: null,
+        major: null,
+        gender: null,
+        age: null,
+      }
+    }
+    
+    this.handleChange = this.handleChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+  
+  handleChange(e) {
+    this.setState({
+      [e.target.name]: e.target.value
+    })
+  }
+  
+  handleSubmit(e) {
+    e.preventDefault()
+    e.stopPropagation()
+    
+    let valid = {
+      name: this.state.name ? true : false,
+      surname: this.state.surname ? true : false,
+      major: this.state.major ? true : false,
+      gender: this.state.gender ? true : false,
+      age: parseInt(this.state.age) ? true : false,
+    }
+    
+    this.setState({
+      valid: valid
+    })
+    
+    if(valid.name && valid.surname && valid.major && valid.gender && valid.age) {
+    
+      fetch("/person", {
+          method: 'POST', // *GET, POST, PUT, DELETE, etc.
+          mode: 'cors', // no-cors, cors, *same-origin
+          cache: 'no-cache', // *default, no-cache, reload, force-cache, only-if-cached
+          credentials: 'same-origin', // include, *same-origin, omit
+          headers: {
+              'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(this.state),
+      })
+      .then(response => response.json())
+      .then(res => {
+        if(res.message) {
```

Comment:
Should you not check the response status code to see if this is an error? Simple checking if there is a field `message` may lead to problems if the successful response contains that field.

Reply:
Of course, you could check the status code, but since changes in the API are not expected and the project is not used differently I don't see much risk with using this kind of if-request

Reply:
I disagree. The project is supposed to be used and maintained in the coming years as well, so we should try to keep it as usable and maintainable as possible. Short-term solutions like this will snowball into bigger problems in the future.